src/screens/Login/Login.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import {
+  TextInput,
+  TouchableHighlight,
+  Text,
+  Image,
+  KeyboardAvoidingView
+} from "react-native";
+import R from "ramda";
+
+import { styles } from "./styles";
+
+const logoImg = require("../../assets/logo.png");
+
+const Login = props => {
+  const { email: emailFromProps, password: passwordFromProps } = props;
+  const [{ email, password }, setLoginData] = useState({
+    email: emailFromProps,
+    password: passwordFromProps
+  });
+
+  const curriedSetLoginData = R.curry((field, value) => {
```

Comment:
could you please clarify why curry is used? in my opinion, this function could be fine without it and perform as intended

Reply:
Using R.curry makes looking this function more succinctly and clearly. Without curring it will be looking like:
field => value => {
    setLoginData(prevData => ({ ...prevData, [field]: value }));
  }

Reply:
Ok, I got it!
In this case, it makes sense to split email and password into separate useState, because currently by using curry you are creating new curriedSetLoginData("email" | "password") for each render, which is not cool.

Reply:
Every render of functional components creates new component after every render (please, see https://overreacted.io/a-complete-guide-to-useeffect/). So I am not fully understand what do you mean.

Reply:
Oh, totally forgot about functional component rerendering with hooks, thank you!
In any case, I would recommend to decompose email and password into different useStates rather than apply carrying. Looks like it's preferred way https://reactjs.org/docs/hooks-state.html#tip-using-multiple-state-variables

Reply:
Updated